File: avancado-intermediario/Java/Q3/Figura.java
public class Figura {
    private double x;
    private double y;

    public Figura(double x, double y) {
        this.x = x;
        this.y = y;
        System.out.println("Criando figura");
    }

    public double getX() {
        return x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        this.y = y;
    }

    public void desenhar() {
        System.out.println("Desenhando uma figura na posição (" + x + ", " + y + ")");
    }

    @Override
    public String toString() {
        return "Figura na posição (" + x + ", " + y + ")";
    }
}

File: avancado-intermediario/Python/Q3/figura.py
class Figura:

    def __init__(self,x:float,y:float):
        print("Criando figura")
        self.__x = x
        self.__y = y

    @property
    def x(self):
        return self.__x

    @x.setter
    def x(self, novo_x:float):
        self.__x = novo_x

    @property
    def y(self):
        return self.__y

    @y.setter
    def y(self, novo_y:float):
        self.__y = novo_y
    
    def desenhar(self):
        print(f"Desenhando uma figura na posição ({self.__x}, {self.__y})")

    def __str__(self) -> str:
        return f"Figura na posição ({self.__x}, {self.__y})"